extension/goplugin/database.go

```diff
@@ -51,17 +52,29 @@ func (db *Database) Clone() *Database {
 	}
 }
 
+func isWithin() bool {
+	pc, _, _, ok := runtime.Caller(4)
```

Comment:
magic number. add a test for this feature in case someone changes something in the future.
Also, according to [the docs ](https://golang.org/pkg/runtime/#FuncForPC) the result may be inaccurate
> If pc represents multiple functions because of inlining, it returns the *Func describing the outermost function.